src/Features/Core/Portable/Shared/Options/ServiceFeatureOnOffOptions.cs

```diff
@@ -13,6 +14,7 @@ internal static class ServiceFeatureOnOffOptions
         /// this option doesn't mean we will show all diagnostics that belong to opened files when turned off,
         /// rather it means we will only show diagnostics that are cheap to calculate for small scope such as opened files.
         /// </summary>
-        public static readonly PerLanguageOption<bool> ClosedFileDiagnostic = new PerLanguageOption<bool>(OptionName, "Closed File Diagnostic", defaultValue: true);
+        public static readonly PerLanguageOption<bool> ClosedFileDiagnostic = new PerLanguageOption<bool>(
+            OptionName, "Closed File Diagnostic", defaultValue: true, perLanguageDefaults: ImmutableDictionary<string, bool>.Empty.Add(LanguageNames.CSharp, false));
```

Comment:
this approach is a bit broken since any language we don't control can't set its default value differently. only what we control can do it.

perfect/flexible solution probably requires a bit more complex solution than this. but not sure whether we really need all that complexity. probably takes simple approach until we have asks.